src/test/groovy/com/google/plugins/ProtobufJavaPluginTest.groovy

```diff
@@ -153,4 +153,91 @@ class ProtobufJavaPluginTest extends Specification {
     where:
     gradleVersion << GRADLE_VERSIONS
   }
+
+  void "test generateCmds should split commands when limit exceeded"() {
+    given: "a cmd length limit and two proto files"
+
+    String baseCmd = "protoc"
+    List<File> protoFiles = [ new File("short.proto"), new File("long_proto_name.proto") ]
+    int cmdLengthLimit = 32
+
+    when: "the commands are generated"
+
+    List<String> cmds = GenerateProtoTask.generateCmds(baseCmd, protoFiles, cmdLengthLimit)
+
+    then: "it splits appropriately"
+    cmds.size() == 2
```

Comment:
One last comment: should be better to assert the actual list content rather than the size.